### PR TITLE
Add more languages

### PR DESCRIPTION
**Pull request by valzav**

### CODE REVIEW

File: src/app/Translator.js [hunk 1]
```diff
@@ -3,18 +3,36 @@ import {connect} from 'react-redux'
...
 import en from 'react-intl/locale-data/en';
 import es from 'react-intl/locale-data/es';
+import ru from 'react-intl/locale-data/ru';
+import fr from 'react-intl/locale-data/fr';
+import it from 'react-intl/locale-data/it';
 import {DEFAULT_LANGUAGE} from 'app/client_config';
 import tt from 'counterpart';
 
-addLocaleData([...en, ...es]);
+addLocaleData([...en, ...es, ...ru, ...fr, ...it]);
```
Comment (goldibex) [inline src/app/Translator.js]: man I love splats

if you wanted to implement splats in algol you'd use the token YADDAYADDA

File: src/app/locales/counterpart/it.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+module.exports = {
+    counterpart: {
+        names: require('date-names/en'),
+        pluralize: require('pluralizers/en'),
+
+        formats: {
+            date: {
+                'default':  '%a, %e %b %Y',
+                long:       '%A, %B %o, %Y',
+                short:      '%b %e'
```
Comment (goldibex) [inline src/app/locales/counterpart/it.js]: this is incorrect. italian short date format is `%e %b.', provided `%b` is the abbreviated month name

File: src/server/app_render.jsx [hunk 3]
```diff
@@ -6,17 +6,34 @@ import universalRender from '../shared/UniversalRender';
...
+const supported_locales = getSupportedLocales();
+
 async function appRender(ctx) {
     const store = {};
     try {
         let login_challenge = ctx.session.login_challenge;
-        let userPreferences = {};
+        let locale = ctx.getLocaleFromHeader();
+        if (locale) locale = locale.substring(0, 2);
+        const locale_is_supported = supported_locales.find(l => l === locale);
```
Comment (goldibex) [inline src/server/app_render.jsx]: camelCase

File: src/app/Translator.js (hunk 1 above)
Comment (valzav) [reply src/app/Translator.js]: You are making me to pronounce ... as YADDAYADDA for the entire life :)

**Comment by originated** [comment]
@goldibex Your requests have been addressed, please comment if any other items or your approval. Thank you.

**Pull request merged**